### DISCUSSION TITLE
How to structure ring networks (if possible)

### DISCUSSION THREAD

**Discussion by maxsteenbergen** [xyflow/xyflow#4303] [Q&A]
Hi all!

We're looking into XYFlow to render some pretty complex industrial flows from producers to consumers. Some of those networks are circular ring networks, with both producers and consumers. Here's a simplified example:

![image](https://github.com/xyflow/xyflow/assets/574278/2c5126a7-b06b-4b0c-b905-c5ceb221324d)

In this example, the outer circles rectangles are nodes, attached with edges to a inner, circular ring edge. The X's are another type of node.

Basically it boils down to connecting edges to other edges, since the ring in this example is practically a bus. Is this possible without injecting nodes at the positions where edges are to be joined (in the example image the white dots on the inner ring)?

> The inner ring does not need to be an actual circle, btw

**Comment by bcakmakoglu** [comment]
You can't actually connect edges to other edges *but* you can use nodes as a connection point to get a similar visual result.

[Here's an example of a "NodeEdge".](https://codesandbox.io/p/sandbox/nodeedge-example-rgdtyk?file=%2Fsrc%2FEdgeNode.tsx)
